### DISCUSSION TITLE
Help a noob understand how the ZigStar UZG-01 and Z2M work together 🙏

### DISCUSSION THREAD

**Discussion by Redsandro** [Koenkk/zigbee2mqtt#21176] [General]
I am new with all this, and I'm trying to understand. Can someone help me figure this out? :sweat_smile: 

I have a brand new UZG-01. It has a Web UI with MQTT settings page. I can connect it to my MQTT server. I thought that was the "Zigbee2MQTT" everyone was talking about. Literally Zigbee in, MQTT out. :monocle_face: 

But apart from getting `state` updates from my UZG-01 over the MQTT server provided, I couldn't figure out how to add Zigbee devices. Turns out I need a Zigbee2MQTT container for that. 

- Is it weird that I keep thinking Z2M should be inside the UZG-01? 
- Do I understand correctly that Z2M is basically a 2nd Web UI to UZG-01?

Before I figured out I need a Zigbee2MQTT container, I ended up with a Home Assistant installation that I thought I could temporarily use just to add devices. But devices added through Home Assistant don't sent MQTT updates.

Do I understand this correctly?:

- The ZHA integration is awesome for HA users, but irrelevant for getting Zigbee signals on a MQTT server
- ZHA and Z2M both create their own private network on the UZG-01 and devices added to one do not show up in the other.
- The UZG-01 doesn't publish to MQTT, but Z2M does.

---

I also never had to authenticate. Both ZHA and Z2M can talk to the UZG-01 without needing a password or an API key. Anyone on the local network can flash the ESP32 or brick the TI chip. Am I missing some obvious hardening?

**Comment by flemmingss** [comment]
I am waiting for my [UZG-01](https://uzg.zig-star.com/), but I can try to give you some answers.

`Is it weird that I keep thinking Z2M should be inside the UZG-01?`
`Do I understand correctly that Z2M is basically a 2nd Web UI to UZG-01?`
Well, The UZG-01 is just an external radio adapter, but with a web gui to control the setup of that radio. I see the use-case to integrate Z2M in it and use it as a all-in-one zigbee hub, but that is not what this product is.

`The ZHA integration is awesome for HA users, but irrelevant for getting Zigbee signals on a MQTT server`
True, ZHA is not a tool to publish to MQTT, it is for getting the devices in Home-Assistant. Zigbee2MQTT is the right tool if you want to translate the zigbee to mqtt. It can be installed as an add-on in Home-Assistant, but it is a custom add-on not included in the official add-ons.

`ZHA and Z2M both create their own private network on the UZG-01 and devices added to one do not show up in the other.`
Yes, because UZG-01 is just the radio-interface.

`The UZG-01 doesn't publish to MQTT, but Z2M does.`
Yes, that is the main point of Zigbee2MQTT (it is in the name, zigbee-to-mqtt)

`I also never had to authenticate. Both ZHA and Z2M can talk to the UZG-01 without needing a password or an API key. Anyone on the local network can flash the ESP32 or brick the TI chip. Am I missing some obvious hardening?`
As I am still waiting for the device I have not looked at this, but I think there is an option to disable the UI at least when connected.
If there is no authentication, yes it may be easy to "hack" from the LAN. I don't think this is a very problematic issue in private homes, but if the network itself is not restricted you can set up a controlled VLAN that other parts of the LAN can't reach,

**Reply by Redsandro** [reply]
Thanks for taking the time to answer my questions. I have only two comments.

> The UZG-01 is just an external radio adapter, but with a web gui to control the setup of that radio. I see the use-case to integrate Z2M in it and use it as a all-in-one zigbee hub, but that is not what this product is.

I still find it a bit weird. The product is like a HAM radio, but the tuning knob is in a VM on a separate device. For its price I'd expect an embedded Pi running Z2M or something. Otherwise why not use a nRF52840 at 1/10th of the price, or flash a CC2520? 

Anyway, once it's set up and forgotten, it works flawlessly.

> `ZHA and Z2M both create their own private network on the UZG-01 and devices added to one do not show up in the other.` Yes, because UZG-01 is just the radio-interface.

Meanwhile, they also answered this question over at [smarthomescene](https://smarthomescene.com/guides/how-to-use-zigbee2mqtt-and-zha-with-a-single-coordinator/). They have some interesting articles on the UZG-01.
